packages/@o3r/rules-engine/src/components/rules-engine/shared/ruleset-history.helper.ts

```diff
@@ -0,0 +1,57 @@
+import type { ActiveRulesetsEvent, AvailableRulesets, DebugEvent, Ruleset, RulesetExecutionErrorEvent, RulesetExecutionEvent } from '../../../engine';
+import type { RulesetExecutionDebug } from '../ruleset-history/ruleset-history-pres.component';
+
+/**
+ * Compute the status of the execution depending on its execution event type, the output and whether the execution
+ * is still active
+ *
+ * @param rulesetExecution
+ * @param isActive
+ */
+export const getStatus = (rulesetExecution: RulesetExecutionErrorEvent | RulesetExecutionEvent, isActive: boolean): 'Error' | 'Active' | 'Deactivated' | 'NoEffect' => {
```

Comment:
can the return type be extracted in an enum? It is reused in `packages/@o3r/rules-engine/src/components/rules-engine/ruleset-history/ruleset-history-pres.component.ts`